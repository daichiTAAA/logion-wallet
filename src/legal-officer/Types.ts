import { ColorTheme, rgbaToHex } from '../component/ColorTheme';
import Identity from '../component/types/Identity';
import PostalAddress from '../component/types/PostalAddress';

export const DEFAULT_LEGAL_OFFICER = "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY"; // Alice
export const ANOTHER_LEGAL_OFFICER = "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty"; // Bob

const LEGAL_OFFICERS: string[] = [ DEFAULT_LEGAL_OFFICER, ANOTHER_LEGAL_OFFICER ];

export function isLegalOfficer(address: string): boolean {
    return LEGAL_OFFICERS.includes(address);
}

export type TokenizationRequestStatus = "PENDING" | "REJECTED" | "ACCEPTED";

export interface AssetDescription {
    assetId: string,
    decimals: number,
}

export interface TokenizationRequest {
    id: string,
    legalOfficerAddress: string,
    requesterAddress: string,
    requestedTokenName: string,
    bars: number,
    status: TokenizationRequestStatus,
    rejectReason?: string | null,
    createdOn?: string,
    decisionOn?: string,
    assetDescription?: AssetDescription,
}

export type LegalOfficerDecisionStatus = "PENDING" | "REJECTED" | "ACCEPTED";

export interface LegalOfficerDecision {
    legalOfficerAddress: string,
    status: LegalOfficerDecisionStatus,
    rejectReason: string | null,
    decisionOn: string | null,
}

export interface ProtectionRequest {
    id: string,
    requesterAddress: string,
    decisions: LegalOfficerDecision[],
    userIdentity: Identity,
    userPostalAddress: PostalAddress,
    createdOn: string
}

export const LIGHT_MODE: ColorTheme = {
    type: 'light',
    shadowColor: rgbaToHex('#3b6cf4', 0.1),
    dashboard: {
        background: rgbaToHex('#3b6cf4', 0.05),
        foreground: '#000000',
    },
    sidebar: {
        background: '#ffffff',
        foreground: '#000000',
        activeItemBackground: rgbaToHex('#3b6cf4', 0.15),
    },
    accounts: {
        iconBackground: '#3b6cf4',
        hintColor: rgbaToHex('#000000', 0.6),
        foreground: '#000000',
        background: rgbaToHex('#3b6cf4', 0.05),
        legalOfficerIcon: {
            category: 'legal-officer',
            id: 'account-shield'
        }
    },
    frame: {
        background: '#ffffff',
        foreground: '#000000',
        link: rgbaToHex('#3b6cf4', 0.15),
    },
    topMenuItems: {
        iconGradient: {
            from: '#3b6cf4',
            to: '#6050dc',
        }
    },
    bottomMenuItems: {
        iconGradient: {
            from: '#7a90cb',
            to: '#3b6cf4',
        }
    }
};
