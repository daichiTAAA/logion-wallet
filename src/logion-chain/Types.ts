import BN from 'bn.js';

export type AssetId = BN;

export type AssetBalance = BN;

export interface AssetMetadata {
    name: string,
    symbol: string,
    decimals: number
}

export interface Asset {
    assetId: AssetId,
    issuer: string,
    metadata: AssetMetadata,
}

export interface AssetWithBalance {
    asset: Asset,
    balance: string,
}
