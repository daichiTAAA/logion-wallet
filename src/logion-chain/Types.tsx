export const enum ReservedName {
    LinkedLocId = "_LINKED_LOC_ID"
}

export interface MetadataItem {
    name: string;
    value: string;
}

export interface LegalOfficerCase {
    owner: string;
    requester: string;
    metadata: MetadataItem[];
    hashes: string[];
    closed: boolean;
    locType: LocType;
}

export type LocType = 'Transaction' | 'Identity';
