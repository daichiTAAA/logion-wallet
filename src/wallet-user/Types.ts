import { ColorTheme, rgbaToHex } from '../component/ColorTheme';

export const DARK_MODE: ColorTheme = {
    type: 'dark',
    shadowColor: rgbaToHex('#3b6cf4', 0.2),
    dashboard: {
        background: '#0c163d',
        foreground: '#ffffff',
    },
    sidebar: {
        background: '#152665',
        foreground: '#ffffff',
        activeItemBackground: rgbaToHex('#ffffff', 0.20),
    },
    accounts: {
        iconBackground: '#3b6cf4',
        hintColor: rgbaToHex('#ffffff', 0.6),
        foreground: '#ffffff',
        background: '#0c163d',
        legalOfficerIcon: {
            category: 'legal-officer',
            id: 'account-shield'
        }
    },
    frame: {
        background: '#152665',
        foreground: '#ffffff',
        link: '#ffffff',
    },
    topMenuItems: {
        iconGradient: {
            from: '#3b6cf4',
            to: '#6050dc',
        }
    },
    bottomMenuItems: {
        iconGradient: {
            from: '#7a90cb',
            to: '#3b6cf4',
        }
    },
    buttons: {
        secondaryBackgroundColor: '#0c163d',
    },
    select: {
        background: rgbaToHex('#ffffff', 0.20),
        foreground: '#ffffff',
        menuBackgroundColor: '#152665',
        selectedOptionBackgroundColor: rgbaToHex('#ffffff', 0.20),
    },
};