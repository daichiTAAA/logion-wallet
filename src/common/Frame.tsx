import React from 'react';

import { Children } from './types/Helpers';
import { ColorTheme } from './ColorTheme';

import './Frame.css';

export interface Props {
    colors: ColorTheme,
    children: Children,
    disabled?: boolean,
    className?: string,
    fullHeight?: boolean,
    altColors?: boolean,
    title?: string,
    fillHeight?: boolean,
}

export default function Frame(props: Props) {

    const backgroundColor =
        (props.altColors !== undefined && props.altColors) ?
            props.colors.frame.altBackground :
            props.colors.frame.background;
    const inlineCss = `
    .Frame a,
    .Frame .btn-link {
        color: ${props.colors.frame.link}
    }
    `;

    let className = "Frame";
    if(props.className !== undefined) {
        className = className + " " + props.className;
    }
    if(props.disabled !== undefined && props.disabled) {
        className = className + " disabled";
    }
    if(props.fullHeight !== undefined && props.fullHeight) {
        className = className + " full-height";
    }
    if(props.fillHeight !== undefined && props.fillHeight) {
        className = className + " fill-height";
    }

    return (
        <div
            className={ className }
            style={{
                backgroundColor,
                color: props.colors.frame.foreground,
                boxShadow: `0 0 25px ${props.colors.shadowColor}`,
            }}
        >
            <style>
            { inlineCss }
            </style>
            {
                props.title !== undefined &&
                <div className="title">{ props.title }</div>
            }
            { props.children }
        </div>
    );
}
