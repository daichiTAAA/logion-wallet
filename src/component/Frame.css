.Frame {
    position: relative;
    padding: 30px;
    border-radius: 10px;
}

.Frame.disabled:before {
    position: absolute;
    content: '';
    top: 0; right: 0; bottom: 0; left: 0;
    z-index: 1;
    background-color: black;
    opacity: 0.3;
    border-radius: 10px;
}
