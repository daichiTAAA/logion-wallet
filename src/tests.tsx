import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
