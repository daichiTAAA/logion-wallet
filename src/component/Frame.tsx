import React from 'react';

import { Children } from './types/Helpers';
import { ColorTheme } from './ColorTheme';

import './Frame.css';

export interface Props {
    colors: ColorTheme,
    children: Children,
}

export default function Frame(props: Props) {

    const inlineCss = `
    .Frame a,
    .Frame .btn-link {
        color: ${props.colors.frame.link}
    }
    `;

    return (
        <div
            className="Frame"
            style={{
                backgroundColor: props.colors.frame.background,
                color: props.colors.frame.foreground,
                boxShadow: `0 0 25px ${props.colors.shadowColor}`,
            }}
        >
            <style>
            { inlineCss }
            </style>
            { props.children }
        </div>
    );
}
