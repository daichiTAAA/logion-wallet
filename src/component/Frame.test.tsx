import { shallowRender } from '../tests';

import Frame from './Frame';
import { LIGHT_MODE } from '../legal-officer/Types';

test("renders", () => {
    const result = shallowRender(
        <Frame
            colors={ LIGHT_MODE }
        >
        </Frame>
    );
    expect(result).toMatchSnapshot();
});
