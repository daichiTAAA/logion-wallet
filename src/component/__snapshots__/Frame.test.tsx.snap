// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="Frame"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "boxShadow": "0 0 25px #3b6cf41a",
      "color": "#000000",
    }
  }
>
  <style>
    
    .Frame a,
    .Frame .btn-link {
        color: #3b6cf426
    }
    .Dashboard .Frame .form-control,
    .Dashboard .Frame .form-control[readonly] {
        background-color: #ffffff;
    }
    
  </style>
</div>
`;
